### PR TITLE
updated tor_reload function to be executed inside run loop

### PR DESCRIPTION
**Pull request by ddeneka-ynd**
running tor_reload from outside the main loop could be leading to unexpected behaviour 

TUN-2852

### CODE REVIEW

**Review by lukewar — APPROVED** [review]
LGTM.

File: scripts/patches/tor-reload.diff [hunk 1]
```diff
@@ -22,27 +12,40 @@ index 123c33a..cdf12ba 100644
...
-@@ -3192,7 +3202,7 @@ tor_cleanup(void)
+ /********* END VARIABLES ************/
+ 
+ /****************************************************************************
+@@ -2368,6 +2372,11 @@ run_main_loop_once(void)
+ 
+   if (nt_service_is_stopping())
+     return 0;
++    
++  if (should_reload == 1) {
```
Comment (mdk) [inline scripts/patches/tor-reload.diff]: To make it 100% bulletproof I'd protect it (the whole reload) with a mutex, something like:
`tor_mutex_acquire(&reload_lock); ... tor_mutex_release(&reload_lock);`

...where `reload_lock` is declared as static mutext in the lines of: 

`static tor_mutex_t *reload_lock = NULL;`

File: scripts/patches/tor-reload.diff [hunk 2]
```diff
@@ -51,14 +54,23 @@ index bd4f7ea..0037e6b 100644
...
                        past. */
-@@ -3523,6 +3533,56 @@ sandbox_init_filter(void)
+@@ -3417,6 +3426,63 @@ sandbox_init_filter(void)
    return cfg;
  }
  
 +void
 +tor_reload(void)
 +{
++  should_reload = 1;
```
Comment (mdk) [inline scripts/patches/tor-reload.diff]: As above, would be great to protect with `tor_mutex`.

**Review by mdk — APPROVED** [review]
Looks good to me, check my comment about adding extra mutex to make it bulletproof.

**Review by mdk — APPROVED** [review]
👍🏻

**Pull request merged**